diff --git a/batch_gen.py b/batch_gen.py
index c8a7428..e18a0f1 100644
--- a/batch_gen.py
+++ b/batch_gen.py
@@ -46,7 +46,7 @@ class BatchGenerator(object):
             batch_input .append(features[:, ::self.sample_rate])
             batch_target.append(classes[::self.sample_rate])
 
-        length_of_sequences = map(len, batch_target)
+        length_of_sequences = list(map(len, batch_target))
         batch_input_tensor = torch.zeros(len(batch_input), np.shape(batch_input[0])[0], max(length_of_sequences), dtype=torch.float)
         batch_target_tensor = torch.ones(len(batch_input), max(length_of_sequences), dtype=torch.long)*(-100)
         mask = torch.zeros(len(batch_input), self.num_classes, max(length_of_sequences), dtype=torch.float)
diff --git a/eval.py b/eval.py
index 2226e42..33489b3 100644
--- a/eval.py
+++ b/eval.py
@@ -36,7 +36,7 @@ def get_labels_start_end_time(frame_wise_labels, bg_class=["background"]):
 def levenstein(p, y, norm=False):
     m_row = len(p)    
     n_col = len(y)
-    D = np.zeros([m_row+1, n_col+1], np.float)
+    D = np.zeros([m_row+1, n_col+1], float)
     for i in range(m_row+1):
         D[i, 0] = i
     for i in range(n_col+1):
@@ -131,8 +131,8 @@ def main():
             fp[s] += fp1
             fn[s] += fn1
             
-    print "Acc: %.4f" % (100*float(correct)/total)
-    print 'Edit: %.4f' % ((1.0*edit)/len(list_of_videos))
+    print("Acc: %.4f" % (100*float(correct)/total))
+    print('Edit: %.4f' % ((1.0*edit)/len(list_of_videos)))
     for s in range(len(overlap)):
         precision = tp[s] / float(tp[s]+fp[s])
         recall = tp[s] / float(tp[s]+fn[s])
@@ -140,7 +140,7 @@ def main():
         f1 = 2.0 * (precision*recall) / (precision+recall)
 
         f1 = np.nan_to_num(f1)*100
-        print 'F1@%0.2f: %.4f' % (overlap[s], f1)
+        print('F1@%0.2f: %.4f' % (overlap[s], f1))
 
 
 if __name__ == '__main__':
diff --git a/model.py b/model.py
index c0cd70b..bfda0db 100644
--- a/model.py
+++ b/model.py
@@ -101,7 +101,7 @@ class Trainer:
             list_of_vids = file_ptr.read().split('\n')[:-1]
             file_ptr.close()
             for vid in list_of_vids:
-                print vid
+                print(vid)
                 features = np.load(features_path + vid.split('.')[0] + '.npy')
                 features = features[:, ::sample_rate]
                 input_x = torch.tensor(features, dtype=torch.float)
@@ -112,7 +112,7 @@ class Trainer:
                 predicted = predicted.squeeze()
                 recognition = []
                 for i in range(len(predicted)):
-                    recognition = np.concatenate((recognition, [actions_dict.keys()[actions_dict.values().index(predicted[i].item())]]*sample_rate))
+                    recognition = np.concatenate((recognition, [list(actions_dict.keys())[list(actions_dict.values()).index(predicted[i].item())]]*sample_rate))
                 f_name = vid.split('/')[-1].split('.')[0]
                 f_ptr = open(results_dir + "/" + f_name, "w")
                 f_ptr.write("### Frame level recognition: ###\n")
